inspektor/lint.py

```diff
@@ -83,11 +99,25 @@ def check_dir(self, path):
 
         :param path: Path to a directory.
         """
+        filenames = []
+        failed_paths = []
         for root, dirs, files in os.walk(path):
             for filename in files:
-                self.check_file(os.path.join(root, filename))
-
-        return not self.failed_paths
+                filename = os.path.join(root, filename)
+                checker = PathChecker(path=filename, args=self.args, label='Lint',
+                                      logger=self.log)
+                if checker.check_attributes('text', 'python', 'not_empty'):
+                    filenames.append(os.path.join(root, filename))
+
+        if filenames:
+            pool = multiprocessing.Pool()
+            opts = self.get_opts()
+            args = [(filename, opts) for filename in filenames]
+            failed_paths = pool.map(run_pylint_file, args)
```

Comment:
an iterator yielding the paths would be probably better (we can deal with lots of paths). Also have you thought about bundling multiple paths to single check? There is a certain overhead when initializing each worker so for loop would be even better (anyway this is still better than no parallelism and it's pretty readable, just asking whether you considered this option)

Reply:
As the titles says it, this is a "quick and dirty" version.

Anyway, as far as alternative approaches are concerned, I'd focus more on delivering a more "universal" solution to inspektor (for other types of checks and that would work when multiple files are given -- this is limited to directory contents).  And, at least initially, I'd try to stick to the multiprocessing utilities and paradigms on the Python standard library.  Just less complexity to deal with, and with the undeniable result benefits.